packages/core/src/store/store.ts

```diff
@@ -392,4 +401,68 @@ export class DefaultAllureStore implements AllureStore, ResultsVisitor {
 
     return allTrs.filter((tr) => tr.environment === env);
   }
+
+  async allTestEnvGroups() {
+    const allTr = await this.allTestResults({ includeHidden: true });
+    const trByTestCaseId = allTr.reduce(
+      (acc, tr) => {
+        const testCaseId = tr?.testCase?.id;
+
+        if (!testCaseId) {
+          return acc;
+        }
+
+        if (acc[testCaseId]) {
+          acc[testCaseId].push(tr);
+        } else {
+          acc[testCaseId] = [tr];
+        }
+
+        return acc;
+      },
+      {} as Record<string, TestResult[]>,
+    );
+
+    return Object.entries(trByTestCaseId).reduce((acc, [testCaseId, trs]) => {
+      if (trs.length === 0) {
+        return acc;
+      }
+
+      const { fullName, name } = trs[0];
+      const envGroup: TestEnvGroup = {
+        id: testCaseId,
+        fullName,
+        name,
+        status: getWorstStatus(trs.map(({ status }) => status)) ?? "passed",
+        testResultsByEnv: {},
+      };
+
+      trs.forEach((tr) => {
+        const env = matchEnvironment(this.#environmentsConfig, tr);
+
+        envGroup.testResultsByEnv[env] = tr.id;
+      });
+
+      return acc.concat(envGroup);
+    }, [] as TestEnvGroup[]);
+  }
+
+  async testEnvGroupByTestCaseId(id: string) {
```

Comment:
Did I miss something or this function isn't used anywhere?

Reply:
Yes, it isn't. Removed the method